File: src/main/java/org/bakkes/game/model/entity/player/invetory/Item.java
package org.bakkes.game.model.entity.player.invetory;

import org.bakkes.game.model.IHasSpriteName;
import org.bakkes.game.model.entity.OverworldEntity;

public class Item extends OverworldEntity implements IHasSpriteName{
	String spriteName;

	@Override
	public String getSpriteName() {
		return spriteName;
	}

	@Override
	public void setSpriteName(final String to) {
		this.spriteName = to;
	}
}

File: src/main/scala/org/bakkes/game/model/entity/player/invetory/Item.scala
package org.bakkes.game.model.entity.player.invetory

import com.google.inject.Inject
import org.bakkes.game.model.entity.OverworldEntity

import scala.beans.BeanProperty

case class  Item  (itemName:String) extends OverworldEntity{
	setName(itemName)
	@Inject()
	def this() = this("")
	@BeanProperty
	var amount:Int = 1
}

